src/Layouts/body/Body.jsx

```diff
@@ -1,55 +1,26 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import GenreFilter from "../../Components/UI/genreFilter/GenreFilter";
 import ResultSort from "../../Components/UI/resultSort/ResultSort";
 import VideosList from "../../Components/VideosList";
-import VidoeModal from '../../Components/UI/videomodal/VideoModal'
-import VideoForm from '../../Components/UI/videoform/VideoForm'
-import MyButton from '../../Components/UI/button/MyButton'
 
-const Body = ({ videos, setVideos }) => {
+const getVideoSorter = (field) => (a, b) => a[field].localeCompare(b[field]);
+
+const Body = ({ videos, setVideos, videoDetail, showEditVideoModal, showDeleteModal }) => {
 
-  const [videoData, setVideoData] = useState();
-  const [editVieoModal, setEditVieoModal] = useState(false);
-  const [deleteModal, setDeleteModal] = useState(false);
   const [sortByField, setSortByField] = useState('');
 
-  const sortVideos = (sort) => {
-    setSortByField(sort);
-    setVideos([...videos].sort((a, b) => a[sort].localeCompare(b[sort])));
-  };
+  useEffect(() => {
+    const videoSorter = getVideoSorter(sortByField);
 
-  const showEditVideoModal = (video, show) => {
-    setVideoData(video);
-    setEditVieoModal(show);
-  };
+    setVideos([...videos].sort(videoSorter));
+  }, [sortByField]);
 
-  const showDeleteModal = (video, show) => {
-    setVideoData(video);
-    setDeleteModal(show);
+  const sortVideos = (sort) => {
+    setSortByField(sort);
   };
```

Comment:
this function does nothing but calls another function with the same set of parameters - so it's kind of redundant. It's better to just use setSortByField directly instead